app-connector-client/test/test_integration.js

```diff
@@ -1,115 +1,189 @@
+
+        await deleteKeysFile()
+
+        await kyma.then(app => {
+            kymaServer = app.listen(port)
+        })
+        await mock("./test/varkes_config.json").then((mock) => {
+            server = express()
+            server.use(mock)
+        })
+
+        await request(server)  //* Make sure we are connected to kyma
+            .post("/connection").send(
+
+                { "url": tokenURL }
+
+            ).set('Accept', 'application/json').
+            expect(200)
+
+    })
+
+    after(() => { //* stop kyma mock after tests
+        kymaServer.close()
+        deleteKeysFile()
+    })
 
     describe('basic routes', function () {
-        it('responds to /', function testSlash(done) {
-            request(server)
+        it('responds to /', function testSlash() {
+            return request(server)
                 .get('/')
-                .expect(200, done);
+                .expect(200);
         });
 
-        it("shows meteadata", done => {
-            request(server)
+        it("shows meteadata", () => {
+            return request(server)
                 .get("/metadata")
-                .expect(200, done)
+                .expect(200)
         })
-        it('404 everything else', function testPath(done) {
-            request(server)
+        it('404 everything else', function testPath() {
+            return request(server)
                 .get('/foo/bar')
-                .expect(404, done);
+                .expect(404);
         });
     });
 
-    describe("Connect to kyma", function () {
-        it("kyma can create certs from token", done => {
-            confURL = process.env.TOKENURL
-            request(server)
-                .post("/connection").send(
-
-                    { "url": confURL }
-
-                ).set('Accept', 'application/json').
-                expect(200).end((err, res) => {
-
-                    !err ? done() : {}
-                })
-        }).timeout(6000)
-    })
-
     describe("api endpoints", () => {
-        var serviceId;
-        it("shows all services at /apis", (done) => {
-            request(server)
-                .get("/")
-                .expect(200, done)
+        it("creates a new service", () => {
+            return createServiceAndReturnId(server)
         })
-        it("handles error when service doesn't exists", (done) => {
-            request(server)
-                .get("/apis/abc-def")
-                .expect(200).end(
-                    (err, response) => {
-                        response.body.error == 404 ? {} : done()
-                    }
-                )
-        })
-        it("creates a new service", (done) => {
-            request(server)
-                .post("/apis/")
-                .send(
-                    fs.readFileSync(serviceMetadata)
+
+        it("deletes a service", async () => {
+            const serviceId = await createServiceAndReturnId(server)

```

Comment:
I think you are using promises in a non-optimal way here. Just chain the promises and return them, the rest will be done by the test framework, no need for async and await keyword